src/client.rs

```diff
@@ -188,6 +189,9 @@ pub struct ClientConfig {
     /// Supported versions, in no particular order.  The default
     /// is all supported versions.
     pub versions: Vec<ProtocolVersion>,
+
+    /// How to verify the server certificate chain.
+    verifier: &'static verify::ServerCertVerifier,
```

Comment:
Good point, that's probably too restrictive. I'll put it in a `Box`.

I don't want to have a lifetime relative to `ClientConfig` because that will leak the need for an explicit lifetime all the way up the object tree into the public API, which IMO is ugly for such an otherwise-heavyweight object.